azure/scope/cluster.go

```diff
@@ -395,7 +398,11 @@ func (s *ClusterScope) SubnetSpecs() []azure.ResourceSpecGetter {
 			SecurityGroupName: azureBastionSubnet.SecurityGroup.Name,
 			RouteTableName:    azureBastionSubnet.RouteTable.Name,
 			Role:              azureBastionSubnet.Role,
-		})
+		}
+		if s.IsVnetManaged() {
```

Comment:
ditto above about maybe checking if there actually are any `azureBastionSubnet.ServiceEndpoints` before potentially assigning a strongly-typed nil value to `subnetSpec.ServiceEndpoints` unnecessarily